### PR TITLE
PWN-6353 - Off ramp. History fixes

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6353

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/history/interactor/stream/AccountStreamSource.kt [hunk 1]
```diff
@@ -62,20 +63,27 @@ class AccountStreamSource(
...
+                userAccountAddress = account.toPublicKey(),
+                before = lastFetchedSignature,
+                limit = batchSize
+            )
+                .map { it.toRpcSignature() }
             lastFetchedSignature = signatures.lastOrNull()?.signature
             buffer.addAll(signatures)
         } catch (e: EmptyDataException) {
-            Timber.tag(TAG).e(e, "Failed to get next HistoryStreamItem because it is empty")
+            Timber.tag(TAG).i(e, "Failed to get next HistoryStreamItem because it is empty")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/interactor/stream/AccountStreamSource.kt:74]: no need to create an entry in Sentry for that, it's not an error

File: app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryPresenter.kt [hunk 2]
```diff
@@ -28,32 +29,10 @@ class TokenHistoryPresenter(
     private val sellTransactionsMapper: HistorySellTransactionMapper,
 ) : BasePresenter<TokenHistoryContract.View>(), TokenHistoryContract.Presenter {
 
-    private class FetchListResult<T>(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryPresenter.kt:31]: extracted to separate file to reuse

File: app/src/main/res/layout/fragment_token_history.xml [hunk 3]
```diff
@@ -105,84 +106,36 @@
...
+            tools:visibility="visible">
 
             <androidx.recyclerview.widget.RecyclerView
                 android:id="@+id/historyRecyclerView"
                 android:layout_width="match_parent"
                 android:layout_height="match_parent"
                 android:clipToPadding="false"
                 android:orientation="vertical"
-                android:paddingTop="24dp"
-                android:paddingBottom="@dimen/history_transaction_list_last_item_padding"
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_token_history.xml:122]: removed paddingBottom with app:layout_constraintBottom_toBottomOf="parent"

File: app/src/main/res/layout/fragment_token_history.xml [hunk 4]
```diff
@@ -105,84 +106,36 @@
...
-            android:layout_height="0dp"
-            android:orientation="vertical"
+            layout="@layout/layout_token_history_error"
             android:visibility="gone"
             app:layout_constraintBottom_toBottomOf="parent"
             app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/view_action_buttons">
-
-            <ImageView
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_token_history.xml:138]: extracted to separate view, you are welcome

**Review by kabramovp2p — APPROVED** [review]

**Review by Davran331122 — APPROVED** [review]

**Pull request merged**